_pkgdown.yml

```
@@ -58,3 +58,4 @@ reference:
     desc: Functions deprecated in teal.data v0.3.1
```

It's better not to specify a version so as to add new entries in the upcoming release (if any)

```suggestion
    desc: Functions deprecated in teal.data
```

---

Perhaps move the version number to the title of the deprecated group of functions `Deprecated TealData class and related functions in teal.data v0.3.1`?

---

if we metion so for the upcoming changes, we will have a separate section for deprecated methods, which will include the version name. At this moment, I am not certain about future deprecated methods, but I suggest not mentioning the version. If we decide in the future, we can keep a consistent heading like "Deprecated Method in v0.3.1" and so on.

---

But, the `TealData` was deprecated in v0.3.1 which is a fact that can't change and we already group all these functions under `TealData`.

---

This information will still be available in documentation of `TealData`.

---

Usually, it's just a title "Deprecated" / "Superseded" and that's it.
Few examples:
https://dplyr.tidyverse.org/reference/index.html#superseded
https://purrr.tidyverse.org/reference/index.html#superseded
https://tidyr.tidyverse.org/reference/index.html#superseded
https://usethis.r-lib.org/reference/index.html#defunct-and-deprecated-functions

It's a convinient solution as you might add to this in the future releases contrary to the situation in which we will be having "Deprecated in 1.0.0", "Deprecated in 2.0.0" and so on. As pointed out, the information of the package number is already available in NEWS and respective function documentation.

---

Is this page expected to be a permanent, possibly growing feature?

---

Permanent - not really. Deprecated functions has to be removed eventually which will make this section empty and probably removed because of that.

Possibly growing - we don't really know, likely yes. That's why I would like to give ourselves some freedom on that regard. But more importantly - I would like to follow some standards from well-known packages outlined above.

---

I would rather avoid the tidyverse's practice of deprecating something every other week, to be honest :wink:

---

LOL. True. This is rather for documenting stuff ;)

---

Here's my point: if this page lists functions deprecated in 0.3.1 and will be removed when those functions are removed (presumably 0.3.3), why not leave 0.3.1 in the title?

---

Because we might add something in 0.3.2. I just don't want to multiply such sections for each of the release for simplicity and convenience for both developers and users. As a developer it's easier (and safer against possible mistakes) for me to add to a single section. As a user - I am actually not really interested from when it's deprecated - I just need to change my code accordingly.

And don't get me wrong - both are technically correct - I just wanted to make life a little bit easier and simpler.

---

I just noticed that the last release was 0.3.0, not 0.3.1.
I believe the plan was to deprecate all of these in one release and drop them in the next, which would be 0.3.2.
Since the changes are huge, we wanted to remove traces rather quickly and invite everyone struggling with migration to the special Github thread. Keeping a list of deprecated and superseded functions in the documentation seems counter to that plan to me.

---

Note that this is a requirement to list all exported objs (including deprecated ones) in the pkgdown. Otherwise pkgdown would complain and we don't want that.

---

Agreed. But we _will_ remove this page very soon, will we not?

---

Yes - as soon as it got empty - 0.3.2 unless new additions will be made